src/basics.md

```diff
@@ -535,3 +576,4 @@ fn main() {
 <!-- Reference -->
 
 [race-condition-file]: https://en.wikipedia.org/wiki/Race_condition#File_systems
+[twitter-hashtag-regex]: https://github.com/twitter/twitter-text/blob/master/java/src/com/twitter/Regex.java#L255
```

Comment:
you can have the link text `[twitter hashtag regex]` without hyphens to avoid link remapping.
also linking to master branch will likely make the reference evaporate. I would suggest linking to specific git revision.